application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -131,6 +143,80 @@ private RestAction<Message> sendExplanationMessage(GuildMessageChannel threadCha
         return action.setEmbeds(embeds);
     }
 
+    /**
+     * Determine between the title of the thread and the first message which to send to the AI. It
+     * uses a simple heuristic of length to determine if enough context exists in a question. If the
+     * title is used, it must also include a question mark since the title is often used more as an
+     * indicator of topic versus a question.
+     * 
+     * @param questionFirstMessage - The first message of the thread which originates from the
+     *        question asker.
+     * @param threadChannel - The thread in which the question was asked.
+     * @return An answer for the user from the AI service or a message indicating either an error or
+     *         why the message wasn't used.
+     */
+    RestAction<Message> constructChatGptAttempt(String questionFirstMessage,
+            ThreadChannel threadChannel) {
+        // Will be at most 100 characters.
+        String questionTitle = threadChannel.getName();
+        StringBuilder stringBuilder = new StringBuilder();
+
+        if (questionFirstMessage.length() < MIN_QUESTION_LENGTH
+                && questionTitle.length() < MIN_QUESTION_LENGTH) {
+            return useChatGptFallbackMessage(threadChannel);
+        }
+
+        if (questionTitle.length() + questionFirstMessage.length() > MAX_QUESTION_LENGTH) {
+            // - 1 for the space that will be added afterward.
+            questionFirstMessage = questionFirstMessage.substring(0,
+                    (MAX_QUESTION_LENGTH - 1) - questionTitle.length());
+        }
+
+        stringBuilder.append(questionTitle).append(" ").append(questionFirstMessage);
+        for (ForumTag tag : threadChannel.getAppliedTags()) {
+            if (stringBuilder.length() > MAX_QUESTION_LENGTH) {
+                break;
+            }
+            stringBuilder.insert(0, String.format("%s ", tag.getName()));
+        }
+
+        String question = stringBuilder.toString();
+        logger.debug("The final question sent to chatGPT: {}", question);
+        Optional<String> chatGPTAnswer = chatGptService.ask(question);
+
+        if (chatGPTAnswer.isPresent()) {
+            String aiResponse = chatGPTAnswer.get();
+            String lowercaseAiResponse = aiResponse.toLowerCase();
```

Comment:
`toLowerCase` must be used with specifying a locale. `toLowerCase(Locale.US)`, otherwise it can result in nasty bugs when we move the bot to a different server.